node/runtime/pangoro/src/bridges_message/pangolin.rs

```diff
@@ -303,3 +307,151 @@ impl SourceHeaderChain<<Self as ChainWithMessages>::Balance> for Pangolin {
+// 	MessageDispatch<AccountIdOf<ThisChain<B>>, BalanceOf<BridgedChain<B>>>
+// 	for DarwiniaFromBridgedChainMessageDispatch<B, ThisRuntime, ThisCurrency,
+// ThisDispatchInstance> where
+// 	BalanceOf<ThisChain<B>>: Saturating + FixedPointOperand,
+// 	ThisDispatchInstance: 'static,
+// 	ThisRuntime: pallet_bridge_dispatch::Config<
+// 			ThisDispatchInstance,
+// 			BridgeMessageId = (LaneId, MessageNonce),
+// 		> + pallet_transaction_payment::Config,
+// 	<ThisRuntime as pallet_transaction_payment::Config>::OnChargeTransaction:
+// 		pallet_transaction_payment::OnChargeTransaction<
+// 			ThisRuntime,
+// 			Balance = BalanceOf<ThisChain<B>>,
+// 		>,
+// 	ThisCurrency: Currency<AccountIdOf<ThisChain<B>>, Balance = BalanceOf<ThisChain<B>>>,
+// 	pallet_bridge_dispatch::Pallet<ThisRuntime, ThisDispatchInstance>:
+// 		bp_message_dispatch::MessageDispatch<
+// 			AccountIdOf<ThisChain<B>>,
+// 			(LaneId, MessageNonce),
+// 			Message = FromBridgedChainMessagePayload<B>,
+// 		>,
+// {
+// 	type DispatchPayload = FromBridgedChainMessagePayload<B>;
+
+// 	fn dispatch_weight(
+// 		message: &DispatchMessage<Self::DispatchPayload, BalanceOf<BridgedChain<B>>>,
+// 	) -> frame_support::weights::Weight {
+// 		message.data.payload.as_ref().map(|payload| payload.weight).unwrap_or(0)
+// 	}
+
+// 	fn dispatch(
+// 		relayer_account: &AccountIdOf<ThisChain<B>>,
+// 		message: DispatchMessage<Self::DispatchPayload, BalanceOf<BridgedChain<B>>>,
+// 	) -> MessageDispatchResult {
+// 		let message_id = (message.key.lane_id, message.key.nonce);
+// 		pallet_bridge_dispatch::Pallet::<ThisRuntime, ThisDispatchInstance>::dispatch(
+// 			B::BRIDGED_CHAIN_ID,
+// 			B::THIS_CHAIN_ID,
+// 			message_id,
+// 			message.data.payload.map_err(drop),
+// 			|dispatch_origin, dispatch_weight| {
+// 				let unadjusted_weight_fee = ThisRuntime::WeightToFee::calc(&dispatch_weight);
+// 				let fee_multiplier =
+// 					pallet_transaction_payment::Pallet::<ThisRuntime>::next_fee_multiplier();
+// 				let adjusted_weight_fee = fee_multiplier.saturating_mul_int(unadjusted_weight_fee);
+// 				if !adjusted_weight_fee.is_zero() {
+// 					ThisCurrency::transfer(
+// 						dispatch_origin,
+// 						relayer_account,
+// 						adjusted_weight_fee,
+// 						ExistenceRequirement::AllowDeath,
+// 					)
+// 					.map_err(drop)
+// 				} else {
+// 					Ok(())
+// 				}
+// 			},
+// 		)
+// 	}
+// }
+
+// Dispatching Bridged -> This chain messages.
+use bridge_runtime_common::messages::AccountIdOf;
+use frame_support::{
+	traits::{Currency, ExistenceRequirement},
+	weights::WeightToFeePolynomial,
+};
+use sp_runtime::{traits::Saturating, FixedPointOperand};
+// use bp_message_dispatch::MessageDispatch;
+use bp_message_dispatch::MessageDispatch as _;
+use bp_messages::target_chain::MessageDispatch;
+use sp_runtime::MultiAddress;
+
+#[derive(RuntimeDebug, Clone, Copy)]
+pub struct FromPangolinMessageDispatch;
```

Comment:
Is this what we expected? Remove complicated generic params, more readable now. @HackFisher